rest/rest.go

```diff
@@ -75,8 +74,8 @@ type corsConfig struct {
 	allowedMethods []string
 }
 
-// CORSConfigOption represents functional-style options to modify the CORS configuration in RunServer.
-type CORSConfigOption func(*corsConfig)
+// ServerConfigOption represents functional-style options to modify the server configuration in RunServer.
+type ServerConfigOption func(*corsConfig, *runtime.ServeMux)
 
 var (
```

Comment:
If you refer to the allowed headers and methods: a `[]string` cannot be a const (for some reason).

Reply:
Yes, lines are off here... Yes, only the port (integer) could be constant. The others can't. One workarround would be to introduce methods like `getDefaultAllowedMethods(){return X}` but the question is if this makes the code more or even less readable.

Since they are all defined as non-constant variables they could also being put into the other `var` block but I guess it is intentional to separate them. Maybe then put a comment before and/or for each Default value to indicate that they are constants.